app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -160,32 +214,57 @@ private void prepareLanguages() {
         CharSequence[] languageNames = languageNameWithCodeList.toArray(new CharSequence[0]);
         CharSequence[] languageCodes = languageCodesList.toArray(new CharSequence[0]);
         // Add all languages and languages codes to lists preference as pair
-        langListPreference.setEntries(languageNames);
-        langListPreference.setEntryValues(languageCodes);
+        descriptionLangListPreference.setEntries(languageNames);
+        descriptionLangListPreference.setEntryValues(languageCodes);
 
         // Gets current language code from shared preferences
-        String languageCode = getCurrentLanguageCode();
-        if (languageCode.equals("")) {
+        String languageCode = getCurrentDescriptionLanguageCode();
+        if (languageCode.equals("")){
+
             // If current language code is empty, means none selected by user yet so use phone local
-            langListPreference.setValue(Locale.getDefault().getLanguage());
+            descriptionLangListPreference.setValue(Locale.getDefault().getLanguage());
         } else {
             // If any language is selected by user previously, use it
-            langListPreference.setValue(languageCode);
+            descriptionLangListPreference.setValue(languageCode);
         }
 
-        langListPreference.setOnPreferenceChangeListener((preference, newValue) -> {
+        descriptionLangListPreference.setOnPreferenceChangeListener((preference, newValue) -> {
             String userSelectedValue = (String) newValue;
-            saveLanguageValue(userSelectedValue);
+            saveDescriptionLanguageValue(userSelectedValue);
             return true;
         });
     }
 
-    private void saveLanguageValue(String userSelectedValue) {
-        defaultKvStore.putString(Prefs.KEY_LANGUAGE_VALUE, userSelectedValue);
+    /**
+     * Changing the default app language with selected one and save it to SharedPreferences
+     */
+    public void setLocale(final Activity activity, final String userSelectedValue) {
+        final Locale locale = new Locale(userSelectedValue);
+        Locale.setDefault(locale);
+        final Configuration configuration = new Configuration();
+        configuration.locale = locale;
+        activity.getBaseContext().getResources().updateConfiguration(configuration,
+            activity.getBaseContext().getResources().getDisplayMetrics());
+
+        final SharedPreferences.Editor editor = activity.getSharedPreferences("Settings", MODE_PRIVATE).edit();
+        editor.putString("language",userSelectedValue);
+        editor.apply();
+    }
+
+    private void saveDescriptionLanguageValue(String userSelectedValue) {
```

Comment:
quick javadoc for all new methods

Reply:
Some previous written similar methods don't have javadocs. Can I also add javadocs for the same? (Although I will try to factorize them)

Reply:
You are right, many methods currently do not have Javadoc, because we introduced this rule recently.
Sure, if you want you can also add Javadoc to methods you modify, that would be very welcome. In the case of totally unrelated methods that would better be left for a separate pull request, though.